### ISSUE TITLE
Change SHR_ASSERT calls to avoid using errMsg

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/CTSM#58]
**Bill Sacks < sacks@ucar.edu > - 2017-02-23 12:13:23 -0700**
**Bugzilla Id:** [2427](http://bugz.cgd.ucar.edu/show_bug.cgi?id=2427)
**Bugzilla CC:**  andre@ucar.edu,  erik@ucar.edu,  rfisher@ucar.edu, 

I have just made a cime PR that allows you to pass file and line information directly to SHR_ASSERT. This avoids the ridiculously large overhead of the shr_log_errMsg call.

https://github.com/ESMCI/cime/pull/1183

Once this version of cime comes into CLM, we should rework the SHR_ASSERT calls to use this new interface rather than using errMsg.

I have started a branch for this here:

https://svn-ccsm-models.cgd.ucar.edu/clm2/branches/assert_file_line

where I have just changed one file for testing purposes.

**Comment by billsacks** [comment]
**Bill Sacks < sacks@ucar.edu > - 2017-02-23 12:13:51 -0700**

This change will significantly speed up CLM in debug mode, but should have little or no impact in production mode.

**Comment by billsacks** [comment]
I have made this change throughout the code (1f4d81c6). However, from looking at total lnd run time before and after this commit, I no longer see a noticeable impact on performance, either with gnu on my mac (tested with `SMS_D_Ld1_P4x1.f10_f10_musgs.I2000Clm50BgcCropQianRsGs.bishorn_gnu.clm-default`; and I did confirm that SHR_ASSERT calls are truly being called by forcing one to fail) or with intel on cheyenne (tested with `SMS_D_Ld3.f10_f10_musgs.I1850Clm50BgcCrop.cheyenne_intel.clm-default`). Looking back at the cime issue referenced above, it looks like I probably saw the performance issue on yellowstone, so maybe the issue was specific to that machine, or maybe I mainly noticed it for single-point runs?

Despite not seeing a benefit, I'll keep this change in, partly because it feels cleaner, and partly in case this still matters for some machines / compilers / configurations.
